### PR TITLE
Adding a task event for setup

### PR DESCRIPTION
**Pull request by diptanu**

### CODE REVIEW

**Comment by dadgar** [comment]
LGTM but need to add docs: https://www.nomadproject.io/docs/http/alloc.html#TaskStates

**Comment by dadgar** [comment]
Fix tests.

File: nomad/structs/structs.go [hunk 1]
```diff
@@ -2769,6 +2769,13 @@ const (
 	// failed.
 	TaskArtifactDownloadFailed = "Failed Artifact Download"
 
+	// TaskBuildingTaskDir indicates that the task direcotry/chroot is being
```
Comment (nanoz) [inline nomad/structs/structs.go]: direcotry  => directory

Comment (dadgar) [inline nomad/structs/structs.go]: Thanks @nanoz! Fixed in master

**Pull request merged**